### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by Drewbrad4**
All done! Forgot an "s" in my branch name but hopefully I can fix that later. Thank you!

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -7,60 +7,120 @@
...
 var temp2 = 350
 var temp3 = 212
 
-
-
+//Tests
+console.log( boilPoint(temp1));
+console.log(boilPoint(temp2));
+console.log(boilPoint(temp3));
+console.log(boilPoint("banana"), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:35]: Arbitrary banana! 

This looks great! Good work! 👍

File: code-challenges.js [hunk 2]
```diff
@@ -7,60 +7,120 @@
...
+  for(let i = 0; i < array.length; i++){
+    newArr.push(array[i] * 5);
+  }
+  return newArr
+}
 
+var myNumbers1 = [3, 7, 0, 6, -9]
 
+// Test
+console.log(multBy5(myNumbers1), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:52]: Excellent!

File: code-challenges.js [hunk 3]
```diff
@@ -7,60 +7,120 @@
...
 
 // --------------------3) Create a function that multiplies each number in the array by 5 using map.
 // Use the test variable provided below. Expected outcome: [40, -35, 0, 30, 10]
 
-var myNumbers2 = [8, -7, 0, 6, 2]
-
-
+const map5 = (array) => {
+  return array.map(value => value * 5);
+}
```
Comment (andeeisaacs) [inline code-challenges.js:59]: Super clean, all on one line!

File: code-challenges.js [hunk 4]
```diff
@@ -7,60 +7,120 @@
...
 
+var myNumbers2 = [8, -7, 0, 6, 2]
 
+// Test
+console.log(map5(myNumbers2), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:64]: Heck yeah. Nice work!

File: code-challenges.js [hunk 5]
```diff
@@ -7,60 +7,120 @@
...
 
 // --------------------4) Create a function that removes all the vowels from a string.
 // Use the test variables provided below. Expected output: "HyThrLrnStdnt" "LvJvScrpt"
 
+const byeVowels = (string) => {
+  let newArr = string.split("")
+  let filteredArr = newArr.filter(value => {
+    return value.toLowerCase() !== "a" && value.toLowerCase() !== "e" && value.toLowerCase() !== "i" && value.toLowerCase() !== "o" && value.toLowerCase() !== "u"
```
Comment (andeeisaacs) [inline code-challenges.js:72]: Excellent use of toLowerCase!

File: code-challenges.js [hunk 6]
```diff
@@ -7,60 +7,120 @@
...
+
 var stringWithVowels1 = "HeyThereLearnStudent"
 var stringWithVowels2 = "ILoveJavaScript"
 
-
-
-
+// Tests
+console.log(byeVowels(stringWithVowels1));
+console.log(byeVowels(stringWithVowels2), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:82]: Fantastic!

File: code-challenges.js [hunk 7]
```diff
@@ -7,60 +7,120 @@
...
 
 // --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
 // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string"
 
+const betterByeVowels = (string) => {
+  if(typeof string === "string"){
+    return string.replace(/[aeiou]/ig, "")
```
Comment (andeeisaacs) [inline code-challenges.js:89]: Ooh, some Regex! Very interesting. I'm wondering how its working here though.

Since we do not teach regex, I would recommend if you ever use it, please pseudo code how it works right above so we can make sure you understand it :)

File: code-challenges.js [hunk 8]
```diff
@@ -7,60 +7,120 @@
...
+  }
+  else {
```
Comment (andeeisaacs) [inline code-challenges.js:91]: The indentation is a little goofy here, the else should be up on the line with the previous curly brace. Lets keep an eye out for that.

File: code-challenges.js [hunk 9]
```diff
@@ -7,60 +7,120 @@
...
 var notAString2 = 42
 
-
-
-
+//Tests
+console.log(betterByeVowels(stringWithVowels1));
+console.log(betterByeVowels(stringWithVowels2));
+console.log(betterByeVowels(notAString1));
+console.log(betterByeVowels(notAString2), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:103]: Great!

File: code-challenges.js [hunk 10]
```diff
@@ -7,60 +7,120 @@
...
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
-var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
-
-
+const onlyCats = (array) => {
+  return array.filter(value => value.animal === "cat")
```
Comment (andeeisaacs) [inline code-challenges.js:109]: Clean!

File: code-challenges.js [hunk 11]
```diff
@@ -7,60 +7,120 @@
...
+}
 
+var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
 
+// Test
+console.log(onlyCats(toonimals), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:115]: Excellent!!! 🎉

File: code-challenges.js [hunk 12]
```diff
@@ -7,60 +7,120 @@
...
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+const nonCats = (array) => {
+  let filteredArr = array.filter(value => value.animal !== "cat")
+  return filteredArr.map(value => value.name)
```
Comment (andeeisaacs) [inline code-challenges.js:122]: This is totally excellent. I think there may be a way you could refactor this to use only the map with a conditional statement inside of its block of code to do the exact same thing. I'd look into this!

File: code-challenges.js [hunk 13]
```diff
@@ -7,60 +7,120 @@
...
+}
+
+//Test
+console.log(nonCats(toonimals), "\n");
```
Comment (andeeisaacs) [inline code-challenges.js:126]: 👏

File: interview-questions.md [hunk 14]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer:
 
+    A function is logic code that can be reused throughout your program to save time and make your life easier.
```
Comment (andeeisaacs) [inline interview-questions.md:12]: 👌 Indeed!

File: interview-questions.md [hunk 15]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+    A function is a set of instructions for the computer to execute. It can take some (optional) inputs and adjust the data and return (optional) outputs when called on to save time when doing repetitive tasks.
+
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
 
+    .map() iterates over every value in an array and performs some action to it. .filter() keeps any value in an array that meets a conditional.
```
Comment (andeeisaacs) [inline interview-questions.md:24]: Heck yeah. Great answer.

File: interview-questions.md [hunk 16]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+    Both return new arrays. .map() performs a function on every value in an array and returns the new adjusted array. .filter() checks every value in an array for a true statement and filters out the false, forming a new array with the filtered values.
+
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
 
+    console.log() is only seen in the terminal or console, it's not generally visible to the user. Return is used within a function to indicate whatever input you want to give back to the user. Functions must have a return.
```
Comment (andeeisaacs) [inline interview-questions.md:36]: Yep! Excellent!

File: interview-questions.md [hunk 17]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+    console.log() displays a parameter in the console window. When a return statement is used, a function will stop executing and return the specified value.
+
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
 
+    An argument is any data the user inputs into the function.
```
Comment (andeeisaacs) [inline interview-questions.md:48]: Yup! But, not just the user! A different function could be passing information into the argument as well. So other parts of your program can also take advantage of the argument.

File: interview-questions.md [hunk 18]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
   Researched answer:
 
+    Input passed to a function.
```
Comment (andeeisaacs) [inline interview-questions.md:52]: Id want to expand a bit more on the researched answer here. Its a bit short! I'm sure theres more cool information you learned about arguments.

File: interview-questions.md [hunk 19]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
 
+    There are two roles. The Driver and the Navigator. The Driver is physically coding on their machine while the Navigator is guiding them through the thought process and looking up any details they may need.
```
Comment (andeeisaacs) [inline interview-questions.md:60]: Yep! I'd also want to include that the roles switch at times.

File: interview-questions.md [hunk 20]
```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+    Predominantly associated with agile development, pair programming is a development technique that involves a Driver and a Navigator. The Driver writes the code and the Navigator observes and provides feedback.
+
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
   Your answer:
 
+    No idea. Best guess would be Top Down Design? Mapping out the big picture before getting into details and features?
```
Comment (andeeisaacs) [inline interview-questions.md:72]: Not quite. But I know you've got it in your researched answer!

File: testing-with-jest.test.js [hunk 21]
```diff
@@ -8,20 +8,62 @@
...
+})
+
+const addThemUp = (num1, num2) => {
+  if(typeof num1 === "number" && typeof num2 === "number"){
+    return num1 + num2;
+  }
+  else {
+    return `Please input a number`;
+  }
+}
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:38]: Awesome!

File: testing-with-jest.test.js [hunk 22]
```diff
@@ -8,20 +8,62 @@
...
+})
+
+const triangleArea = (num1, num2) => {
+  if(typeof num1 === "number" && typeof num2 === "number"){
+    return num1 * num2 / 2;
+  }
+  else {
+    return `Please input a base and height`;
+  }
+}
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:69]: Fantastique!

**Review by andeeisaacs — APPROVED** [review]
Drew! Excellent first week assessment. Your code was clean and worked great! 

My advice for you is:

- I noticed you didn't pseudo code anything. Pseudo coding is super important to communicate your work well and allow people to follow your thought process as you were writing your code. As it stands right now, if I look at your code I can see how it works, but I have no idea how you came to the conclusions to use what you did in order to achieve your solutions.  Remember, coding is ALL about communication, not just to yourself, but to other people looking at your code as well. 

- You made choices to drop your working variables below your functions. This is a big no-no. This is something called Hoisting. If we understand how code reads, it reads from top down. If your function that needs a variable passed into it is below your function, it essentially has to stop what its doing, search everywhere on the document for the variable being passed into it and then hoist it up to the correct place then run your function. 

While right now your code works, because javascript is doing extra leg work to make it happen, this will cause a lot of problems later in life. Best practice is to leave any working variables that a function will take in, ABOVE the function. 

You can read more about it here: https://www.jenniferbland.com/how-hoisting-in-javascript-can-cause-unexpected-results/

Otherwise you're doing excellent! Please merge to master!
